### ISSUE TITLE
unable to setup solaris_exporter

### ISSUE THREAD

**Issue by Umedha76** [n27051538/solaris_exporter#14]
followed all the instructions but when I tryied to launch solaris_exporter by
python2.7 solaris_exporter.py

> Traceback (most recent call last):
>   File "solaris_exporter.py", line 133, in <module>
>     from prometheus_client.core import REGISTRY, Counter, Gauge, GaugeMetricFamily, CounterMetricFamily, UntypedMetricFamily
>   File "/usr/lib/python2.7/site-packages/prometheus_client/__init__.py", line 3, in <module>
>     from . import (
>   File "/usr/lib/python2.7/site-packages/prometheus_client/exposition.py", line 113
>     def make_wsgi_app(registry: CollectorRegistry = REGISTRY, disable_compression: bool = False) -> Callable:
>                               ^
> SyntaxError: invalid syntax

root@solaris-test:/opt/solaris_exporter# uname -a
SunOS solaris-test 5.11 11.4.0.15.0 i86pc i386 i86pc

Can some one help me to get this running.

thanks

**Comment by n27051538** [comment]
Hello, 
Use the latest README. 

 pip2.7 install prometheus_client==0.7.1
        # Note than current versions of prometheus_client (0.14.1 +) for some reasons not works with Python 2.7, 
        # that is why I fixed its version to 0.7.1, it is tested.

**Comment by Umedha76** [comment]
> Hello, Use the latest README.
> 
> pip2.7 install prometheus_client==0.7.1 # Note than current versions of prometheus_client (0.14.1 +) for some reasons not works with Python 2.7, # that is why I fixed its version to 0.7.1, it is tested.

Thank you so much for the prompt respond
let me check

**Comment by Umedha76** [comment]
> > Hello, Use the latest README.
> > pip2.7 install prometheus_client==0.7.1 # Note than current versions of prometheus_client (0.14.1 +) for some reasons not works with Python 2.7, # that is why I fixed its version to 0.7.1, it is tested.
> 
> Thank you so much for the prompt respond let me check

Thank you its working now.
++++++++++++++++++++++++++++++++++
but when I tried to set it up as a service I get the following messages and do not see the output on web like I gout it earlier

```

> root@solaris-test:/opt/solaris_exporter# ./solaris_exporter_smf.sh install
> ./solaris_exporter_smf.sh Script for adding solaris_exporter.py to Solaris SMF as prometheus/solaris_exporter
>      r    Removing "Prometheus::solaris:cmd:::/usr/sbin/fmadm:uid=0" from  /etc/security/exec_attr
>      r    Removing "Prometheus::solaris:cmd:::/usr/sbin/nvmeadm:uid=0" from  /etc/security/exec_attr
>      r    Removing "Prometheus::solaris:cmd:::/usr/sbin/raidctl:uid=0" from  /etc/security/exec_attr
>      r    Removing "Prometheus::solaris:cmd:::/usr/sbin/zlogin:uid=0" from  /etc/security/exec_attr
>      r    Removing "Prometheus:::User could run some monitoring commands" from /etc/security/prof_attr
>      r    Current profiles of monitor user is:
> SMF prometheus/solaris_exporter removed
> Installing solaris_exporter.py as SMF prometheus/solaris_exporter
>     i
> [ 2022 Sep  4 02:24:04 svc.startd could not set context for method:  ]
> chdir: No such file or directory
> [ 2022 Sep  4 02:24:04 Stopping because all processes in service exited. ]
> [ 2022 Sep  4 02:24:04 Executing start method ("/usr/bin/python2.7 /opt/solaris_exporter/solaris_exporter.py"). ]
> [ 2022 Sep  4 02:24:04 Environment variable "LC_ALL=" already set. ]
> [ 2022 Sep  4 02:24:04 svc.startd could not set context for method:  ]
> chdir: No such file or directory
> online          2:24:04 svc:/prometheus/solaris_exporter:default
> root@solaris-test:/opt/solaris_exporter#
> root@solaris-test:/opt/solaris_exporter#
> root@solaris-test:/opt/solaris_exporter#
> root@solaris-test:/opt/solaris_exporter#
> root@solaris-test:/opt/solaris_exporter#
> root@solaris-test:/opt/solaris_exporter#
> root@solaris-test:/opt/solaris_exporter# svcs solaris_exporter
> STATE          STIME    FMRI
> online          2:24:48 svc:/prometheus/solaris_exporter:default

```

Kind regards,

**Comment by Umedha76** [comment]
after creating the /exporter/home/monitor directory and setting its permission to user monitor
the service configured and it is working.

BTW will this same setup work in solaris 11.3.35.6.0  (0.5.11)
